hail/python/test/hailtop/utils/test_gcs_requester_pays.py

```diff
@@ -0,0 +1,27 @@
+from operator import attrgetter
+from pytest import raises
+
+from hailtop.aiotools.router_fs import RouterAsyncFS
+from hailtop.fs.router_fs import RouterFS
+from hailtop.utils import async_to_blocking
+from hailtop.utils.gcs_requester_pays import gcs_requester_pays_fs_cache
+
+
+def test_get_fs_by_requester_pays_config():
+    def test_with_cls(cls, kwargs_field):
+        config_1 = "foo"
+        config_2 = ("foo", ["bar", "baz", "bat"])
+        kwargs_1 = {"gcs_requester_pays_configuration": config_1}
+        kwargs_2 = {"gcs_requester_pays_configuration": config_2}
+        fses = gcs_requester_pays_fs_cache(cls)
+        assert attrgetter(kwargs_field)(fses[None]) == {}
+        assert attrgetter(kwargs_field)(fses[config_1]) == kwargs_1
```

Comment:
Interesting, I've never seen `attrgetter` before. I think I would prefer though adding a `_gcs_kwargs` property to both `RouterFS` and `RouterAsyncFS` and accessing that property directly, so that if I mess something up the linters will tell me that these attr accesses will fail without me having to run this test.

Reply:
added!